server/src/main/java/org/opensearch/index/store/remote/utils/TransferManager.java

```diff
@@ -153,7 +159,11 @@ public IndexInput getIndexInput() throws IOException {
 
         @Override
         public long length() {
-            return request.getLength();
+            long length = 0;
+            for (BlobFetchRequest.BlobPart blobPart : request.blobParts()) {
```

Comment:
This is better suited as a method of `BlobFetchRequest`. We can calculate it once at object creation / when parts are added to the request and simply return the value here.

Reply:
Makes sense, I will update this.

Reply:
Addressed in latest commit